kratos/geometries/brep_surface.h

```diff
@@ -117,6 +118,22 @@ class BrepSurface
     {
     }
 
+    // Constructor for SBM
+    BrepSurface(
+        typename NurbsSurfaceType::Pointer pSurface, 
+        BrepCurveOnSurfaceLoopArrayType& BrepOuterLoopArray,
+        BrepCurveOnSurfaceLoopArrayType& BrepInnerLoopArray,
+        ModelPart& rSurrogateModelPartInner, ModelPart& rSurrogateModelPartOuter)
+        : BaseType(PointsArrayType(), &msGeometryData)
+        , mpNurbsSurface(pSurface) 
+        , mOuterLoopArray(BrepOuterLoopArray)
+        , mInnerLoopArray(BrepInnerLoopArray)
+        , mpSurrogateModelPartInner(&rSurrogateModelPartInner)
+        , mpSurrogateModelPartOuter(&rSurrogateModelPartOuter)
```

Comment:
This is the extra constructor I was referring to. As far as I see `mpSurrogateModelPartInner` and `mpSurrogateModelPartOuter` are not being used. Most probably I'm missing something.